dbms/src/AggregateFunctions/AggregateFunctionAvg.h

```diff
@@ -117,12 +119,13 @@ class AggregateFunctionAvg final
 
     void insertResultInto(ConstAggregateDataPtr __restrict place, IColumn & to, Arena *) const override
     {
-        if constexpr (IsDecimal<TResult>)
+        if constexpr (IsDecimal<CalculateType>)
         {
             ScaleType left_scale = result_scale - scale;
-            TResult result = this->data(place).sum.value * getScaleMultiplier<TResult>(left_scale)
-                / static_cast<typename TResult::NativeType>(this->data(place).count);
-            static_cast<ColumnDecimal<TResult> &>(to).getData().push_back(result);
+            CalculateType result = this->data(place).sum.value * getScaleMultiplier<CalculateType>(left_scale)
+                / static_cast<typename CalculateType::NativeType>(this->data(place).count);
+            static_cast<ColumnDecimal<TResult> &>(to).getData().push_back(
+                TResult(static_cast<typename TResult::NativeType>(result.value)));
```

Comment:
```suggestion
            CalculateType::NativeType result = this->data(place).sum.value * getScaleMultiplier<CalculateType>(left_scale)
                / static_cast<typename CalculateType::NativeType>(this->data(place).count);
            static_cast<ColumnDecimal<TResult> &>(to).getData().push_back(
                TResult(static_cast<typename TResult::NativeType>(result)));
```
I think there is no need to use `CalculateType` as the middle type?